modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1757,9 +1764,31 @@ private Map<String, Serializable> _toObjectValues(
 					objectField.getDBType(),
 					ObjectFieldConstants.DB_TYPE_DATE)) {
 
-				values.put(
-					objectField.getName(),
-					_toDate(locale, String.valueOf(value)));
+				String dateValue = String.valueOf(value);
+
+				if (!StringUtil.equals(objectField.getName(), "createDate") &&
+					!StringUtil.equals(objectField.getName(), "modifiedDate")) {
+
+					String pattern = ObjectFieldUtil.getDateTimePattern(
```

Comment:
It seems like a lot of code to do a common thing, I imagine other teams have had this need before, so maybe these classes/methods below can help you write less code. Can you please try this?
`DateUtil, DateFormatFactoryUtil, GetterUtil#getDate(java.lang.String, java.text.DateFormat)`
This [link](https://www.baeldung.com/java-date-without-time) can help too.

Another point is that maybe the right solution is to override `ObjectFieldBusinessType#getValue` method in `DateObjectFieldBusinessType` class and move this logic to there, what do you think?
Maybe we can submit this specific fix in another pull request so as not to block others and also verify the impact of this change by running tests only with it.